lib/fs.js

```diff
@@ -132,7 +132,7 @@ function makeCallback(cb) {
   }
 
   return function() {
-    return cb.apply(null, arguments);
+    return cb(...arguments);
```

Comment:
OK, I went back to `.apply(undefined, arguments)` which probably won't correct as many edge cases, but is better than the current `.apply(null, arguments)` in that regard and doesn't have a negative perf hit.